src/Chapter2.hs

```diff
@@ -841,7 +873,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate x l
+  | x < 0 = []
+  | otherwise = take (length l) $ drop x (cycle l)
+-- rotate x l = drop x l ++ take x l
```

Comment:
Nice catch! This is also a correct solution, but it would be less efficient.
Also, you may be interested in [`splitAt`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:splitAt) 😉